lib/primitives/tx.js

```diff
@@ -1041,16 +1042,25 @@ class TX {
     const addrs = [];
 
     for (const output of this.outputs) {
-      const addr = output.getAddress();
+      if (output.script.isMultisig()) {
+        const [, n] = output.script.getMultisig();
 
-      if (!addr)
-        continue;
+        for (let i = 1; i <= n; i++) {
+          const keyring = KeyRing.fromMultisigScript(output.script, i);
+          table.add(keyring.getHash());
```

Comment:
Mine as well leave in the `addrs` part of the return object as well, right? The non-multisig fork here just does `output.getAddress()` (pretty simple) but that would return `null` if an address was multisig. Looks like that array is only used by `tx.getOutputAddresses()` which itself isn't used anywhere outside of testing, but it would fail if a tx had a multisig address. In fact, maybe you should include this diff to `tx-test.js` in your PR:


```diff
diff --git a/test/tx-test.js b/test/tx-test.js
index 0a83eab7..f48712f5 100644
--- a/test/tx-test.js
+++ b/test/tx-test.js
@@ -800,7 +800,8 @@ describe('TX', function() {
     // by vouts, not merged.
     const outputAddresses = [
       Address.fromBase58('1fLeMazoEy8FfgeFcppRxNYZs54jyLccw'),
-      Address.fromBase58('1EeREnzQujX7CLgmzDSaebS48jxjeyBHQM')
+      Address.fromBase58('1EeREnzQujX7CLgmzDSaebS48jxjeyBHQM'),
+      Address.fromBase58('3GVXYDpGkqukE5qey1YYf4ykGUfPYq6Zja')
     ];

     const getOutputAddresses = tx.getOutputAddresses();
```

Reply:
Added `keyring.getAddress()` to `addrs`.

`tx.getOutputAddresses()` is not only used in testing. It is used by `wdb.getWalletsByTX(tx)` which is in turn used to retrieve which `wallet`s are interested in the `tx`.